README.md

```diff
@@ -77,6 +77,25 @@ If you want to create a package with all the dependencies bundled, you can run `
 
 Every patch is expected to adhere to our coding style, which is checked automatically by Travis. You can install the checkers locally using `npm run install-dependencies`, and then run the checks using `npm run check` before submitting a pull request. There is also `npm run fix`, that will attempt to fix the formatting.
 
+## Dockerizing selfoss
+
+Dockerfile for Selfoss RSS aggregator is bundled in the repository.
```

Comment:
It is not clear to me from this text who this is directed for. Is this for:

1. People who want to run production instances of selfoss but want it isolated from other apps on their system. — Docker might not be the best way to do that but perhaps it is the most convenient one.
2. People who want to run production selfoss and do not want to bother with installing web server and PHP (they do not need to bother with composer and NPM as we ship prebuilt archives) – this looks like the most probable option. 
3. Developers who do not want to bother with installing web server, composer and NPM — fair but developers will probably also need to run `npm run build` or `composer install` when they change something, is there a way to do that without rebuilding the whole image? Also `npm run dev`.
4. Developers that want uniform development environment – that does not make much sense since selfoss is expected to run at wide variety of web servers and PHP runtimes.

Reply:
The `README` should explain which of these use cases are targeted and how to accomplish their goals.

Reply:
> It is not clear to me from this text who this is directed for. Is this for:
> 
> 1. People who want to run production instances of selfoss but want it isolated from other apps on their system. — Docker might not be the best way to do that but perhaps it is the most convenient one.
> 2. People who want to run production selfoss and do not want to bother with installing web server and PHP (they do not need to bother with composer and NPM as we ship prebuilt archives) – this looks like the most probable option.
> 3. Developers who do not want to bother with installing web server, composer and NPM — fair but developers will probably also need to run `npm run build` or `composer install` when they change something, is there a way to do that without rebuilding the whole image? Also `npm run dev`.
> 4. Developers that want uniform development environment – that does not make much sense since selfoss is expected to run at wide variety of web servers and PHP runtimes.

Running selfoss on docker is on user's choice.
If there's some person who wants to run selfoss on docker then probably he/she will know their use case.
 I think defining/writing it will be too fffortful and time consuming as well.

Reply:
> It is not clear to me from this text who this is directed for. Is this for:
> 
> 1. People who want to run production instances of selfoss but want it isolated from other apps on their system. — Docker might not be the best way to do that but perhaps it is the most convenient one.
> 2. People who want to run production selfoss and do not want to bother with installing web server and PHP (they do not need to bother with composer and NPM as we ship prebuilt archives) – this looks like the most probable option.
> 3. Developers who do not want to bother with installing web server, composer and NPM — fair but developers will probably also need to run `npm run build` or `composer install` when they change something, is there a way to do that without rebuilding the whole image? Also `npm run dev`.
--It may be tricky and we'll have to work on this usecase that is, if the source gets updated then use npm run build or npm run dev.
> 4. Developers that want uniform development environment – that does not make much ...

Reply:
`npm run dev` and other dev steps should be used together with the dev docker image - I was planning to submit it once I have it cleaned up, you guys are too fast :-D

For the dev image we will just need to install php, apache, composer and npm all together in one image and then mount the source directory with a volume - the image itself will not contain any source code from selfoss, only all the tools that devs might use. I just didn't have enough time to test it, and installing npm together with php is a pain...

Reply:
> I'm actually not that experienced in docker.
>  Now I too feel that I did it so early.
> You seem to have in depth knowledge of docker, and if you are willing to take it completely, then can I remove this pull request entirely?
> And I feel that I'll be able to learn a lot from you. :-)
> `npm run dev` and other dev steps should be used together with the dev docker image - I was planning to submit it once I have it cleaned up, you guys are too fast :-D
> 
> For the dev image we will just need to install php, apache, composer and npm all together in one image and then mount the source directory with a volume - the image itself will not contain any source code from selfoss, only all the tools that devs might use. I just didn't have enough time to test it, and installing npm together with php is a pain...

Reply:
Depends. We could transform this PR into the dev docker image, and then have another one for the production image. I'll send you the Dockerfile soon once I finish dealing with node user, grrr.. (can I create a PR against your PR? I've never done that before...) But we definitely want to keep the entrypoint in utils because even the production image will depend on it.

Reply:
> seems the good idea.
> sure, you can create the pull request against this. Can we do that? I don't know that how to acomplish it or whether it is doable or not.
:-)
> Depends. We could transform this PR into the dev docker image, and then have another one for the production image. I'll send you the Dockerfile soon once I finish dealing with node user, grrr.. (can I create a PR against your PR? I've never done that before...) But we definitely want to keep the entrypoint in utils because even the production image will depend on it.